crates/ty_python_semantic/src/types/signatures.rs

```diff
@@ -456,6 +459,262 @@ impl<'db> Signature<'db> {
             .is_some_and(|return_type| return_type.is_fully_static(db))
     }
 
+    /// Creates a new version of this signature with its generic context specialized to match
+    /// another signature's generic context. This is used for assignability/subtyping checks
+    /// between two generic function signatures where the type variable identity shouldn't matter.
+    fn with_specialized_generic_context<'a>(
+        &'a self,
+        db: &'db dyn Db,
+        other: &'a Signature<'db>,
+    ) -> std::borrow::Cow<'a, Signature<'db>> {
+        if let (Some(left_gc), Some(right_gc)) = (self.generic_context, other.generic_context) {
```

Comment:
Same here, it's useful to have "self" and "other" in this context instead of "left" and "right" as we can easily infer which generic context this belongs to.